bot-review/src/index.ts

```diff
@@ -0,0 +1,63 @@
+import { getInput, info, setFailed, warning } from '@actions/core'
+import { getOctokit } from '@actions/github'
+
+async function run() {
+  try {
+    // defaults
+    const [owner, component] = process.env['GITHUB_REPOSITORY'].split('/')
+    const branch =
+      process.env['GITHUB_HEAD_REF'] || process.env['GITHUB_REF'].slice(11)
+
+    // inputs
+    const ghToken = getInput('gh-token') || process.env['GITHUB_TOKEN']
+    const pull_request = parseInt(getInput('pull'), 10)
+    const repo = getInput('component') || component
+    const event = getInput('event')
+
+    // Assertions
+    if (
+      !ghToken ||
+      !['APPROVE', 'COMMENT', 'REQUEST_CHANGES'].includes(event)
+    ) {
+      throw new Error('A required argument is missing')
+    }
+
+    // Initialize GH client
+    const octokit = getOctokit(ghToken)
+
+    // get most recent PR of current branch
```

Comment:
Only needed if `pull_request` is not set?

Reply:
uh yeah, I should probably separate that into a function then